added invoice receipts

psousa opened the pull request.

psousa committed:
added invoice receipts

psousa committed:
fix wrong method name in send email for invoice receipt

paulosilva86 commented:
:ok: Still missing tests for InvoiceReceipts though right?

psousa commented:
Yes, didn't have time for those. Probably by the end of the week.

paulosilva86 commented:
Ok. No problem. Good job. You can merge whenever you want.

vascosantos commented:
Hi! Any chance of completing this merge anytime soon?

igorescobar commented:
Any chance of completing this merge anytime soon?

igorescobar commented:
@donbonifacio would be great to merge this. this is actually missing in the implementation.

igorescobar commented:
https://invoicexpress.com/api/invoices/partial-payment too

igorescobar commented:
@psousa can you take a look into this conflicts?

vascosantos commented:
@igorescobar I cloned this repo onto my own account and implemented the invoice receipts part (which is the part I was missing for my project). I did not include any tests.
Seems to be working fine so far.
Feel free to use from: https://github.com/vascosantos/invoicexpress-gem

igorescobar commented:
@vascosantos cool but @paulosilva86 gave the green light to merge this and somehow It wasn't...

vascosantos commented:
@igorescobar, I saw that as well, but the merge has been pending for more than 1 year. So I looked at the diffs in the code and copy&pasted on the repository I sent you.

Commit:
added rate limit

Commit:
organizing core gem files

Commit:
improving credit notes and invoices. adding more tests and fixtures. see changelog

Commit:
fixed double owner_invoice_id parameter sent

Commit:
bumped version to 0.2.5

Commit:
added transport guides. changed params_encoder to flat. see changelog. fixed some params on update invoice

Commit:
Added city field to client

Commit:
Merge branch 'master' of github.com:weareswat/invoicexpress-gem

Commit:
Merge branch 'master' of github.com:weareswat/invoicexpress-gem

Commit:
Merge branch 'master' of github.com:psousa/invoicexpress-gem

Commit:
Added current_invoice_receipt_number and  current_invoice_receipt_sequence_id to sequences. Bumped version.

Commit:
Fix Travis ruby 2.1 error in transport guide class

igorescobar commented:
👓

psousa commented:
I didn't receive notifications from these comments.

So 3 things:
- I've added tons of missing features/endpoints which were missing from the API and I needed over the last year. Things like transport guides or using logo in email. 
- I've bumped the version (sorry, sorry, sorry, I know I should't have done).
- It doesn't have new specs for everything. I won't have time to add them.

I hope further pull requests will be done in a faster, more orderly way.

I'll paste below the changes I've made:

# Version 0.3.2
Added current_invoice_receipt_number and  current_invoice_receipt_sequence_id to sequences.      

# Version 0.3.1.1
Merged changes from other contributors. Bumped version.

# Version 0.3.0
More big changes on this version:
- added logo option to send email
- added transport guides client
- set params_encoder = Faraday::FlatParamsEncoder in GET requests. We need this for the way that IX accepts arrays in GET requests. See https://github.com/lostisland/faraday/issues/78
- fixed missing owner_invoice_id, manual_sequence_number, currency_code, rate params on update credit note and debit note
- added models for address, message, transport guide
- added tests for transport guides

# Version 0.2.5
Big changes on this version:
- updated send email endpoint for invoices
- cleanup gemspec file
- started cleaning up test files from should to expect
- added manual_sequence_number field to invoices. Required for non portuguese accounts with manual sequence numbering.
- added saft_hash field to invoices
- added currency, multicurrency to invoices. not supported officially.
- added owner_invoice_id to credit_note. not working.

# Version 0.2.0
Support for 429 RateLimitExceeded code
Added Invoice Receipt
Added supplier model
